tests/integration.rs

```diff
@@ -57,10 +60,7 @@ fn sweep(args: &[&str]) -> Command {
 }
 
 /// Sets the target folder and runs the given command
-fn run<'a>(mut cmd: impl BorrowMut<Command>, target: impl Into<Option<&'a Path>>) -> Assert {
-    if let Some(target) = target.into() {
-        cmd.borrow_mut().env("CARGO_TARGET_DIR", target);
-    }
```

Comment:
> Tests should be easy to follow step-by-step by reading the test function, but folks have tried to be too clever with the code.

I wasn't "trying to be clever", `Into<Option<T>>` is just some way for you to provide an argument without `Some()` :smiley:.

---

Also, I don't see any clear reason for reverting this, the `golden_reference` test is the only one that doesn't require a `target`, because it returns at `--help` and `--version` (nothing is done), so it makes sense to explicitly say "This runs with no target". All other tests expect some target path so you just provide it.

But well, it's just a style change so it kinda doesn't matter what I think about it.

Reply:
Hi @marcospb19 I wasn't primarily thinking about the `run` function, but more the whole setup of the tests. For example, the fact that the test modifies the `CARGO_TARGET_DIR` at all is weird to me.  I didn't want to criticize anyone in particular, sorry 🙏

My tests instead copy the whole project into a temp folder and runs it there, which gives the following advantages:

* It isn't necessary to give the `target` directory at all.
* Tests are more realistic, with regards to how `cargo-sweep` actually works.
* Bugs that affect the project files can be found.

But to answer your specific question, I'd say that the `run` function is an abstraction over running _something_, not necessarily the `sweep` command, so including the target directory as a parameter to it would make it weird if someone would like to use `run` to run other commands.

One good example is the `build` function that also just assumes that the build must put target files in a temporary directory, which is so weird to me. It was used in too many places so I didn't want to change it.

Overall I'd say that my feeling is that if all tests were changed to use the same test methodology `recursive_multiple_root_workspaces` (which doesn't use the `build` function), they would be easier to read, understand, and also later modify when future requirements change.

In short: Tests _should be verbose_ 🥰

Reply:
> For example, the fact that the test modifies the CARGO_TARGET_DIR at all is weird to me.

Using a temporary directory seems okay, but in that case please explicitly unset it so the test still pass if I have CARGO_TARGET_DIR set locally.

Reply:
Thanks @jyn514 I have updated the `recursive_multiple_root_workspaces` test to ignore `CARGO_TARGET_DIR`